app/controllers/api/v1/posts_controller.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    # Controller class that is responsible for handling post requests.
+    class PostsController < ActionController::API
+      include DeviseTokenAuth::Concerns::SetUserByToken
+      include Pundit
+
+      before_action :authenticate_user!, except: %i[index show]
+      before_action :set_post, except: %i[index create]
+      before_action :authorize_post!
+      after_action :verify_authorized
+
+      # GET /posts or /posts.json
+      def index
+        @posts = Post.all
+        render json: @posts, status: :ok
+      end
+
+      # GET /posts/:id or /posts/:id.json
+      def show
+        render json: @post, status: :ok
+      end
+
+      # POST /posts or /posts.json
+      def create
+        if post_params.blank?
```

Comment:
we may move these(in `create` and `update` methods) conditions to `before_action`

Reply:
We can get rid of this completely.
Anyway, we validate `if post.save`